consensus/propagation/Reactor.go

```diff
@@ -135,3 +149,713 @@ func (blockProp *Reactor) Receive(chID byte, peer p2p.Peer, msgBytes []byte) {
 		Message:   uw,
 	})
 }
+
+// getPeer returns the peer state for the given peer. If the peer does not exist,
+// nil is returneblockProp.
+func (blockProp *Reactor) getPeer(peer p2p.ID) *PeerState {
+	blockProp.mtx.RLock()
+	defer blockProp.mtx.RUnlock()
+	return blockProp.peerstate[peer]
+}
+
+// getPeers returns a list of all peers that the data routine is aware of.
+func (blockProp *Reactor) getPeers() []*PeerState {
+	blockProp.mtx.RLock()
+	defer blockProp.mtx.RUnlock()
+	peers := make([]*PeerState, 0, len(blockProp.peerstate))
+	for _, peer := range blockProp.peerstate {
+		peers = append(peers, peer)
+	}
+	return peers
+}
+
+// setPeer sets the peer state for the given peer.
+func (blockProp *Reactor) setPeer(peer p2p.ID, state *PeerState) {
+	blockProp.mtx.Lock()
+	defer blockProp.mtx.Unlock()
+	blockProp.peerstate[peer] = state
+}
+
+// ProposeBlock is called when the consensus routine has created a new proposal
+// and it needs to be gossiped to the rest of the network.
+func (blockProp *Reactor) ProposeBlock(proposal *types.Proposal, haves *bits.BitArray) {
+	blockProp.HandleProposal(proposal, blockProp.self, haves)
+}
+
+// HandleProposal adds a proposal to the data routine. This should be called any
+// time a proposal is recevied from a peer or when a proposal is created. If the
+// proposal is new, it will be stored and broadcast to the relevant peers.
+func (blockProp *Reactor) HandleProposal(proposal *types.Proposal, from p2p.ID, haves *bits.BitArray) {
```

Comment:
I think we could actually include the compact block in the proposal in a non-breaking way if we just add a field that isn't hashed or signed over (so literally just don't touch the signBytes or hash functions, just add the field to the prototype)

this way, peers that are running new reactor will send us proposals with compact blocks, which is nice and we get both in the state at the same time. peers not running the new proposal will just ignore the field in those proposals